### PR TITLE
docs: added product categories conceptual

### PR DESCRIPTION
**Pull request by shahednasser**
Added conceptual guide for product categories with the help of @riqwan

Closes DOCS-649

### CODE REVIEW

File: docs/content/modules/products/categories.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Products can be available in more than one category. You can then filter products by a single or list of product categories.
+
+The relation is implemented in the Product entity. You can access the product categories a product is available in by expanding the `categories` relation and using `product.categories`
+
+### ProductCategory
+
+The parent-child hierarchy is represented in the `ProductCategory` entity through the following relations:
+
+- You can access the parent of a category by expanding the `parent_category` relation and accessing `category.parent_category`. You can also access the ID of the parent category by accessing `category.parent_category_id`. If a category doesn’t have a parent, the relation and the ID will be `null`.
+- You can access the children of a category by expanding the `category_children` relation and accessing `category.category_children`. If a category doesn’t have children, the relation will be an empty array.
```
Comment (riqwan) [inline docs/content/modules/products/categories.md:73]: Worth it to note here that without `include_descendant_tree=true`, the `category_children` attribute will only return immediate children. with the flag, you will get the entire descendant nodes.

Comment (shahednasser) [reply docs/content/modules/products/categories.md:73]: Is this flag passed to the endpoints or where exactly?

Comment (riqwan) [reply docs/content/modules/products/categories.md:73]: yes, its a flag passed to the endpoint

Comment (riqwan) [reply docs/content/modules/products/categories.md:73]: Just checked, its in there. I spelt it wrong here 😅 `include_descendants_tree`

**Pull request merged**